src/commands/init.rs

```diff
@@ -99,28 +183,49 @@ pub fn init(cookbook_executable: &str, mut args: Peekable<Iter<String>>) -> Resu
             ))
         }
     };
-    let _shell_type = match args.next() {
+    let shell_env = match env::var("SHELL") {
```

Comment:
We can't rely on `$0` because it's undefined in fish shell, so fallback to `$SHELL`

Reply:
Alternatively, I like requiring people in the invocation to use ``` tome init sc ./example fish | source```. That's entirely reasonable and has precedent in direnv, for example.

Reply:
yeah, I think it's more robust to have the invoker pass the shell, rather than implicitly assume some environment variable.

That's how it was originally, correct? you just pass in the shell type during init.

Reply:
See here: https://github.com/toumorokoshi/tome/pull/4/files#diff-cadd09612a7d662d0ed12de2a0c70dc7afbea48e3aa5861398fea744dc49f476R188

After my change, the logical flow is:
1. Try the existing lookup of `$0` which is the args.next() on L188.
2. If Some, use it
3. If None, then fallback to using `$SHELL` env var which is automatically set in terminal :).

I didn't drop your existing way :D. But I did add a fallback for when that would result in an error with fish.

But but but... see my longer comment on PR for my overall suggestion that hews closer to what you're asking for and adds commandline interface consistency.

Reply:
(Weird, this was folded strangely, so I responded directly to your comment in a void 🥼)

Reply:
ah ok, I see what you did. It sounds like you're planning on switching to always passing in a shell argument which is good. If you want to merge this in and address that in a future PR, that's fine too.

Reply:
👍 ok, I'll do this in the future PR.